bot/src/trader.rs

```diff
@@ -46,7 +44,16 @@ impl Trader {
             Channel::Listed | Channel::PriceChanged => {
                 if let (Some(mean), Some(price)) = (stats.mean_price, data.price) {
                     if Self::is_mean_reliable(&stats) && (price as f64) < BUY_THRESHOLD * mean {
-                        self.handle_purchase(&data, mean).await?;
+                        let list = self.http.fetch_market_data(data.skin_id, 0).await?;
+                        let mut id_lowest = data.id;
+                        let mut price_lowest = price;
+                        for market_data in &list {
+                            if market_data.price < price as f64 {
+                                id_lowest = market_data.id.clone(); 
```

Comment:
..and so you wouldn't have to `clone()` (deep copy) the `String`, you can just reference the String: `&market_data.id`